exec/vm.go

```diff
@@ -340,6 +340,16 @@ func (vm *VirtualMachine) AddAndCheckGas(delta uint64) bool {
 	return true
 }
 
+// Reset makes de VM ready for a new call to Execute if the previous interruption
+// has been interrupted.
+func (vm *VirtualMachine) Reset() {
+	vm.CurrentFrame = -1
+	vm.Exited = false
+	vm.ExitError = nil
+	vm.CallStack = make([]Frame, DefaultCallStackSize)
```

Comment:
If CallStack is already of the right len/cap it would be nice to avoid allocating a new one, and simply clear the existing one